pkg/config/file_test.go

```diff
@@ -20,78 +20,180 @@ package config
 
 import (
 	"context"
+	"fmt"
+	"io/fs"
 	"reflect"
 	"testing"
 	"time"
 
-	"github.com/caas-team/sparrow/pkg/checks/runtime"
-
 	"github.com/caas-team/sparrow/pkg/checks/health"
+	"github.com/caas-team/sparrow/pkg/checks/runtime"
+	"github.com/caas-team/sparrow/pkg/config/test"
+	"gopkg.in/yaml.v3"
 )
 
 func TestNewFileLoader(t *testing.T) {
 	l := NewFileLoader(&Config{Loader: LoaderConfig{File: FileLoaderConfig{Path: "config.yaml"}}}, make(chan runtime.Config, 1))
 
-	if l.path != "config.yaml" {
-		t.Errorf("Expected path to be config.yaml, got %s", l.path)
+	if l.config.File.Path != "config.yaml" {
+		t.Errorf("Expected path to be config.yaml, got %s", l.config.File.Path)
 	}
 	if l.cRuntime == nil {
 		t.Errorf("Expected channel to be not nil")
 	}
+	if l.fsys == nil {
+		t.Errorf("Expected filesystem to be not nil")
+	}
 }
 
 func TestFileLoader_Run(t *testing.T) {
-	type fields struct {
-		path string
-		c    chan runtime.Config
-	}
-	type args struct {
-		ctx    *context.Context
-		cancel *context.CancelFunc
-	}
-
 	tests := []struct {
-		name   string
-		fields fields
-		args   args
-		want   runtime.Config
+		name      string
+		config    LoaderConfig
+		want      runtime.Config
+		expectErr bool
```

Comment:
Nitpicking, but since we're using `wantErr` as a variable everywhere else, please use `wantErr` here as well :) Please replace all occurrences.